google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/cloudpubsub/PublisherSettings.java

```diff
@@ -190,10 +201,14 @@ public com.google.cloud.pubsublite.internal.Publisher<MessageMetadata> newPublis
                 .setPartition(partition)
                 .setStreamFactory(
                     responseStream -> {
-                      ApiCallContext context =
+                      GrpcCallContext context =
                           getCallContext(
                               PubsubContext.of(framework()),
                               RoutingMetadata.of(topicPath(), partition));
+                      if (enableCompression()) {
+                        context =
```

Comment:
In CPS we don't just blindly enable compression and instead use a threshold based on the payload size as load tests showed that enabling it for small payloads was not gaining anything but actually hurting the CPU utilization.

Reply:
We can't do that since our RPCs are actually streams.